daffodil-japi/src/main/scala/org/apache/daffodil/japi/Daffodil.scala

```diff
@@ -141,7 +141,6 @@ class Compiler private[japi] (private var sCompiler: SCompiler) {
   ): ProcessorFactory = {
 
     val pf = sCompiler.compileFile(schemaFile, Option(rootName), Option(rootNamespace))
-    pf.isError
     new ProcessorFactory(pf)
```

Comment:
Lines 143 - 144 look strange at first glance.  We get a pf which is a ProcessorFactory and wrap another ProcessorFactory around it.  I do realize that pf is actually a core Daffodil ProcessorFactory class and the ProcessorFactory on line 144 is actually a Java API ProcessorFactory class.  Nevertheless, I wish we could avoid using these Java and Scala API wrapper classes if possible.

Reply:
Yeah, I really want to get rid of this wrapper stuff. It's complicated to maintain and provides very little real benefit. Pretty much the only real difference is sometimes the Scala API returns a `Seq` but the Java API returns a `List`. Getting rid of this would be a good candidate for a 4.0.0 release as well.

Reply:
I think the pattern for getting rid of this wrapper stuff is to have a java-usable abstract base/interface defined in a ".java" file, that has all the methods the Java API user would need to call including static factory methods. This ".java" file also has all the javadoc. 

Then the scala implementation derives its actual classes from that base/interface, and those can all use scala types and such. Then the "API" for java users is just returning the object, upcasted to the java base/interface that shows only the java-usable methods. 

This doesn't prevent the Java user from just downcasting and using something internal, but I think the whole wrapper approach, which does prevent the downcasting bypass, is just not worth the maintenance headache. Plus this is open source, so what are we really preventing anyway?